src/algorithms/dictator-sort.ts

```diff
@@ -0,0 +1,24 @@
+import SortingAlgorthim from "@contracts/sorting-algorithm";
+
+export default class DictatorSort<
+  T extends number | string
+> extends SortingAlgorthim<T> {
+  constructor(collectionToSort: T[], sortDesc = false) {
+    super(collectionToSort, sortDesc);
+  }
+
+  sort(): T[] {
+    const sortedCollection = [...this.collectionToSort];
+    for (let i = 1; i < this.collectionToSort.length; i++) {
+      // for each element in the collection
```

Comment:
You don't need comments this granular. Unless there is complicated logic, there is no need to comment on built-in features of the language